src/python/pants/backend/go/target_types.py

```diff
@@ -53,6 +53,23 @@ class GoCgoEnabledField(TriBoolField):
     )
 
 
+class GoRaceDetectorEnabledField(TriBoolField):
+    """Enables the Go data race detector."""
+
+    alias = "with_race"
```

Comment:
`race_detector=True`?

Reply:
The corresponding Go option is just `race`. Maybe this should just match the Go name for it?

Reply:
I was leaning away from that name in order to be clearer, but matching now I'm leaning back toward just using the Go name. Bazel rules_go also uses `race` for the name.